portfolio/src/main/webapp/script.js

```diff
@@ -109,30 +108,90 @@ function getQuotes() {
-      quoteListElement.innerHTML = '';
-
-      Object.values(quotesJson).forEach(
-        (quote) => {
-          quoteListElement.appendChild(createQuotesListElement(quote)); 
-        });        
+    .then((responseJson) => {
+      updateLoginStatusElements(responseJson);
+      if (responseJson['Quote'] !== undefined) {
+        const quotesJson = JSON.parse(responseJson['Quote']);
+        createQuotesList(quotesJson);
+      }      
     });
 }
 
 /**
- * This is a helper function, called inside getQuotes. 
- * This function creates a li element that contains each individual quote;
+ * Creates a list for all fetched quotes from the GET request. 
+ */
+function createQuotesList(quotesJson) {
+  if (quotesJson !== null) {
+    const quoteListElement = document.getElementById('quoteWrapper');
+    quoteListElement.innerHTML = '';
+
+    Object.values(quotesJson).forEach(
+      (quote) => {
+        quoteListElement.appendChild(createQuotesListElement(quote)); 
+      }); 
+  }
+}
+
+/**
+ * Creates a li element that contains each individual quote;
  * this li element will be added to the Quotes section in the HTML file.  
  */
 function createQuotesListElement(quote) {
   const liElement = document.createElement('li');
-  liElement.innerText = quote.text;
+  const nickname = quote.nickname;  
+
+  if (nickname === undefined) {
+    liElement.innerText = quote.text.concat(' -- unknown user');
+  } else {
+    liElement.innerText = quote.text.concat(' -- ').concat(nickname);
+  }
   return liElement;
 }
 
 /**
- * This function performs a POST request to deleteQuote servlet, which 
+ * Updates divs and texts on the UI according to the user's login status. 
+ */
+function updateLoginStatusElements(responseJson) {
+  const showAfterLogin = document.getElementById('showAfterLogin'); 
+  const loginText = document.getElementById('loginText');
+  const loginStatus = document.getElementById('loginStatus'); 
+  const redirectText = document.getElementById('redirectText');
+  redirectText.href = responseJson['redirectUrl'];
+
+  const status = responseJson['loggedIn'];
+  if (status === 'true') {
+    showAfterLogin.style.visibility = 'visible';
+    loginText.style.visibility = 'hidden';
+    redirectText.innerText = 'Log out ';
+    handleNicknameDisplay(responseJson);  
+  } else {
+    showAfterLogin.style.visibility = 'hidden';
+    loginText.style.visibility = 'visible';
+    loginStatus.innerText = 'Not logged in yet';
+    redirectText.innerText = 'Log in ';    
+  }     
+}
+
+/**
+ * Displays the username's nickname if a nickname is stored in the response
+ * Json returned by GET request. If a nickname does not exist,
+ * the textbox prompting the user to create a nickname appears. 
+ */
+function handleNicknameDisplay(responseJson) {
+  const nickname = responseJson['nickname'];
+  const createNicknameText = document.getElementById('createNickname'); 
+
+  if (nickname === undefined || nickname === null) {
```

Comment:
I think in Javascript it might be equivalent to,

if (!nickname) {
  ...
}

But please double check before simplifying the logic this way.

Reply:
Yes, the ! symbol works in this case.